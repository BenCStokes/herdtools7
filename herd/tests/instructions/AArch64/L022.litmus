AArch64 L022
(* Intra-thread subroutine calls *)
{

}
  P0          | P1           ;
 MOV W0,#4    | ADR X1,Ldec  ;
 BL Linc      | MOV W0,#4    ;
 BL Linc      | BLR X1       ;
 B Lzero      | BLR X1       ;
Linc:         | B Lone       ;
 ADD W0,W0,#1 |Ldec:         ;
 RET          | SUB W0,W0,#1 ;
Lzero:        | RET          ;
              |Lone:         ;
forall 0:X0=6 /\ 1:X0=2
