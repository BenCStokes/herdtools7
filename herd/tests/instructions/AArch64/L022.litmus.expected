Test L022 Required
States 1
0:X0=6; 1:X0=2;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=6 /\ 1:X0=2)
Observation L022 Always 1 0
Hash=63978aad883ca6ecd3f15a9b500fce5f

